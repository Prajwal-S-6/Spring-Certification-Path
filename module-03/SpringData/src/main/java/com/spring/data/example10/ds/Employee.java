package com.spring.data.example10.ds;


import java.sql.Date;

public record Employee(int id, String firstName, String lastName, String email, String phoneNumber, Date hireDate, float salary) {
}
