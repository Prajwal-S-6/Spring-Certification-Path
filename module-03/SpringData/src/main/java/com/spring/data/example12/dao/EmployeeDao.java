package com.spring.data.example12.dao;

import org.springframework.stereotype.Repository;

@Repository
public class EmployeeDao {
}
