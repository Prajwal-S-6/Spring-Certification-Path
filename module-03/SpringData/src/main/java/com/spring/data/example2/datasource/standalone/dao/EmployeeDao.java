package com.spring.data.example2.datasource.standalone.dao;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.jdbc.core.JdbcTemplate;
import org.springframework.stereotype.Repository;

import javax.sql.DataSource;
import java.util.List;

@Repository
public class EmployeeDao {
    private final JdbcTemplate jdbcTemplate;

    public EmployeeDao(DataSource dataSource) {
        this.jdbcTemplate = new JdbcTemplate(dataSource);
    }

    public List<String> findEmployeeEmails() {
        return jdbcTemplate.queryForList("SELECT EMAIL FROM EMPLOYEE", String.class);
    }
}
