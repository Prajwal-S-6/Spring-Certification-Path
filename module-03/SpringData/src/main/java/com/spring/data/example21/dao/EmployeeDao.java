package com.spring.data.example21.dao;

import org.springframework.data.jpa.repository.JpaRepository;

public interface EmployeeDao extends JpaRepository<EmployeeDao, Integer> {
}
