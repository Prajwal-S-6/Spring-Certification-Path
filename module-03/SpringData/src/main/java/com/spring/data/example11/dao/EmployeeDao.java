package com.spring.data.example11.dao;

import com.spring.data.example11.ds.Employee;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.jdbc.core.JdbcTemplate;
import org.springframework.stereotype.Repository;

import javax.sql.DataSource;

@Repository
public class EmployeeDao {

    private JdbcTemplate jdbcTemplate;

    @Autowired
    public void setDataSource(DataSource dataSource) {
        jdbcTemplate = new JdbcTemplate(dataSource);
    }

    public void saveEmployee(Employee employee) {
        if(employee.id() < 0) {
            throw new IllegalArgumentException("EmployeeId cannot be less than 0");
        }
        int numOfRecordsInserted = jdbcTemplate.update("insert into employee values(?, ?, ?, ?, ?, ?, ?)", employee.id(), employee.firstName(), employee.lastName(), employee.email(), employee.phoneNumber(), employee.hireDate(), employee.salary());
        if (numOfRecordsInserted == 1)
            System.out.println(String.format("Saved employee [%d]", employee.id()));
        else
            throw new IllegalStateException(String.format("Expected 1 record to be inserted, instead retrieved [%d] number of records inserted", numOfRecordsInserted));
    }
}
