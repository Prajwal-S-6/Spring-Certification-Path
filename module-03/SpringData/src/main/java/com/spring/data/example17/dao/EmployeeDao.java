package com.spring.data.example17.dao;

import com.spring.data.example17.ds.Employee;
import org.springframework.beans.factory.annotation.Autowired;

import org.springframework.jdbc.core.JdbcTemplate;
import org.springframework.stereotype.Repository;

import javax.sql.DataSource;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.List;

@Repository
public class EmployeeDao {
    private JdbcTemplate jdbcTemplate;

    @Autowired
    public void setDataSource(DataSource dataSource) {
        jdbcTemplate = new JdbcTemplate(dataSource);
    }

    public List<Employee> findAllEmployeesByIdRange(int minId, int maxId) {
        return jdbcTemplate.query(
                "select employee_id, first_name, last_name, email, phone_number, hire_date, salary " +
                        "from employee " +
                        "where employee_id between ? and ?",
                new Object[]{minId, maxId},
                this::mapEmployee
        );
    }

    public void saveEmployee(Employee employee) {
        int numberOfRecordsInserted = jdbcTemplate.update(
                "insert into employee(employee_id, first_name, last_name, email, phone_number, hire_date, salary) " +
                        "values (?, ?, ?, ?, ?, ?, ?)",
                employee.id(),
                employee.firstName(),
                employee.lastName(),
                employee.email(),
                employee.phoneNumber(),
                employee.hireDate(),
                employee.salary()
        );

        if (numberOfRecordsInserted == 1)
            System.out.println(String.format("Saved employee [%d]", employee.id()));
        else
            throw new IllegalStateException(String.format("Expected 1 record to be inserted, instead retrieved [%d] number of records inserted", numberOfRecordsInserted));
    }

    public void maskEmail(int id) {
        int numberOfUpdatesRows = jdbcTemplate.update(
                "update employee set email='XXX.XXX@XXX.XXX' where employee_id = ?",
                id
        );

        if (numberOfUpdatesRows == 1)
            System.out.println(String.format("Updated employee email [%d]", id));
        else
            throw new IllegalStateException(String.format("Expected 1 record to be updated, instead updated [%d] number of records inserted", numberOfUpdatesRows));
    }

    private Employee mapEmployee(ResultSet resultSet, int i) throws SQLException {
        return new Employee(
                resultSet.getInt("employee_id"),
                resultSet.getString("first_name"),
                resultSet.getString("last_name"),
                resultSet.getString("email"),
                resultSet.getString("phone_number"),
                resultSet.getDate("hire_date"),
                resultSet.getFloat("salary")
        );
    }
}
