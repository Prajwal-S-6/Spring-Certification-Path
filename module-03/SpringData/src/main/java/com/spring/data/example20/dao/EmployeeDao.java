package com.spring.data.example20.dao;

import com.spring.data.example20.ds.Employee;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.jdbc.core.JdbcTemplate;
import org.springframework.jdbc.core.RowMapper;
import org.springframework.stereotype.Repository;

import javax.sql.DataSource;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.List;

@Repository
public class EmployeeDao {

    private JdbcTemplate jdbcTemplate;

    @Autowired
    public void setDataSource(DataSource dataSource) {
        jdbcTemplate = new JdbcTemplate(dataSource);
    }
    public List<Employee> findAllEmployees() {
        return jdbcTemplate.query("select * from employee", this::mapEmployee);
    }

    private Employee mapEmployee(ResultSet resultSet, int rowNum) throws SQLException {
        return new Employee(
                resultSet.getInt("employee_id"),
                resultSet.getString("first_name"),
                resultSet.getString("last_name"),
                resultSet.getString("email"),
                resultSet.getString("phone_number"),
                resultSet.getDate("hire_date"),
                resultSet.getFloat("salary")
        );
    }

    public void saveEmployee(Employee employee) {
        if(employee.id() < 0) {
            throw  new IllegalArgumentException("Employee id cannot be less than or equal to zero");
        }

        int numOfRecordsInserted = jdbcTemplate.update("insert into employee values(?,?,?,?,?,?,?)",
                employee.id(),
                employee.firstName(),
                employee.lastName(),
                employee.email(),
                employee.phoneNumber(),
                employee.hireDate(),
                employee.salary());
        if (numOfRecordsInserted == 1)
            System.out.println(String.format("Saved employee [%d]", employee.id()));
        else
            throw new IllegalStateException(String.format("Expected 1 record to be inserted, instead retrieved [%d] number of records inserted", numOfRecordsInserted));
    }

    public void deleteAllEmployees() {
        int numOfRowsDeleted = jdbcTemplate.update("delete from employee");
        System.out.println(String.format("Employees deleted, number of deleted rows = [%d]", numOfRowsDeleted));
    }
}
