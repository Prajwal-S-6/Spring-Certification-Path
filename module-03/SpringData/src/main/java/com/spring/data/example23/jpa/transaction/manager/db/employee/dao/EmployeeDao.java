package com.spring.data.example23.jpa.transaction.manager.db.employee.dao;

import com.spring.data.example23.jpa.transaction.manager.db.employee.ds.Employee;
import org.springframework.data.jpa.repository.JpaRepository;

public interface EmployeeDao extends JpaRepository<Employee, Integer> {
}
