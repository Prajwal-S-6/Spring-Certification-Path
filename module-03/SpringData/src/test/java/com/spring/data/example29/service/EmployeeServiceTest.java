package com.spring.data.example29.service;

import com.fasterxml.jackson.core.JsonProcessingException;
import com.fasterxml.jackson.databind.ObjectWriter;
import com.fasterxml.jackson.databind.util.JSONPObject;
import com.spring.data.example29.dao.EmployeeDao;
import com.spring.data.example29.ds.Employee;
import org.approvaltests.Approvals;
import org.json.JSONArray;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase;
import org.springframework.boot.test.autoconfigure.orm.jpa.AutoConfigureDataJpa;
import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
import org.springframework.boot.test.context.SpringBootTest;
import com.fasterxml.jackson.databind.ObjectMapper;

import java.util.List;

import static org.assertj.core.api.Assertions.assertThat;
import static org.junit.jupiter.api.Assertions.*;

@SpringBootTest
@AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.ANY)
class EmployeeServiceTest {

    @Autowired
    private EmployeeService employeeService;

    @Autowired
    private EmployeeDao employeeDao;

    ObjectMapper objectMapper;
    ObjectWriter writer;

    @BeforeEach
    void setUp() {

        objectMapper = new ObjectMapper();
        writer = objectMapper.writerWithDefaultPrettyPrinter();
    }

    @Test
    void shouldSaveAllEmployees() throws JsonProcessingException {

        employeeService.saveAllEmployees();

        List<Employee> employeeList = employeeDao.findAll();

        assertThat(employeeList.size()).isEqualTo(5);
        Approvals.verify(writer.writeValueAsString(employeeList));
    }

    @Test
    void shouldQueryWillowZhangEmployee() {
        Employee employee = employeeService.queryEmployee();

        assertThat(employee.getFirstName()).isEqualTo("Willow");
        assertThat(employee.getLastName()).isEqualTo("Zhang");
    }
}