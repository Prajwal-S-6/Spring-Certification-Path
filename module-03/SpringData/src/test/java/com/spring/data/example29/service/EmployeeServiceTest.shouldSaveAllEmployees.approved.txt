[ {
  "id" : 1,
  "firstName" : "John",
  "lastName" : "Doe",
  "email" : "John.Doe@corp.com",
  "phoneNumber" : "555-55-55",
  "hireDate" : 1559673000000,
  "salary" : 70000.0
}, {
  "id" : 2,
  "firstName" : "Willow",
  "lastName" : "Zhang",
  "email" : "Willow.Zhang@corp.com",
  "phoneNumber" : "555-55-56",
  "hireDate" : 1562869800000,
  "salary" : 80000.0
}, {
  "id" : 3,
  "firstName" : "Jayvon",
  "lastName" : "Krueger",
  "email" : "Jayvon.Grant@corp.com",
  "phoneNumber" : "555-55-57",
  "hireDate" : 1563301800000,
  "salary" : 90000.0
}, {
  "id" : 4,
  "firstName" : "Shaylee",
  "lastName" : "Mcclure",
  "email" : "Shaylee.Mcclure@corp.com",
  "phoneNumber" : "555-55-58",
  "hireDate" : 1563474600000,
  "salary" : 120000.0
}, {
  "id" : 5,
  "firstName" : "Miley",
  "lastName" : "Krueger",
  "email" : "Miley.Krueger@corp.com",
  "phoneNumber" : "555-55-59",
  "hireDate" : 1563561000000,
  "salary" : 110000.0
} ]